cmd/installer/pkg/install/errata.go

```diff
@@ -0,0 +1,33 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package install
+
+import (
+	"log"
+	"os"
+
+	pkgkernel "github.com/siderolabs/talos/pkg/kernel"
+	"github.com/siderolabs/talos/pkg/machinery/kernel"
+)
+
+// errataBTF handles the case when kexec from pre-BTF kernel to BTF enabled kernel always fails.
+//
+// This applies to upgrades of Talos < 1.3.0 to Talos >= 1.3.0.
+func errataBTF() {
+	_, err := os.Stat("/sys/kernel/btf/vmlinux")
+	if err == nil {
+		// BTF is enabled, nothing to do
+		return
+	}
+
+	log.Printf("disabling kexec due to upgrade to the BTF enabled kernel")
+
+	if err = pkgkernel.WriteParam(&kernel.Param{
+		Key:   "proc.sys.kernel.kexec_load_disabled",
+		Value: "1",
+	}); err != nil {
+		log.Printf("failed to disable kexec: %s", err)
```

Comment:
why not return all the way?

Reply:
I don't want to fail on upgrade e.g. if someone compiles a custom kernel with kexec disabled altogether.

I can't always rely on the presence of this field. Another might be upgrading from such an old version of Talos that it doesn't have kexec. So don't fail the upgrade if possible :)

Reply:
got it, makes sense